src/outputtype.h

```diff
@@ -27,6 +28,8 @@ enum class OutputType {
     CHANGE_AUTO,
 };
 
+const std::array<OutputType, 3> OUTPUT_TYPES = {OutputType::LEGACY, OutputType::P2SH_SEGWIT, OutputType::BECH32};
```

Comment:
Doesn't defining the value of non-primitive constants in the header file potentially cause linker issues in C++ (or duplication, as this literal data will be included in multiple files)?. I think it's better to only declare the variable here then define the value in the `cpp`.

Reply:
I don't think this would cause any issues since it's const. I also haven't seen any linker warnings or errors about this. I think there would only be some duplication, but isn't that the same for any constant global variable that is defined in a header?

Reply:
If you are fine trusting the word of [Microsoft](https://docs.microsoft.com/en-us/cpp/cpp/const-cpp?view=vs-2019),

> In C, constant values default to external linkage, so they can appear only in source files. In C++, constant values default to internal linkage, which allows them to appear in header files.

Another source said C++ does not normally create storage for consts, which I assume means they work like #defines, in which case there shouldn't be wasted duplicate data entries either.

Reply:
If you replace the {...} initialized with a function call, and then have the constant in multiple translation units, the function will get called once for each.

Reply:
I moved the initialization to the cpp file